testing/mock_test.ts

```diff
@@ -126,6 +126,17 @@ Deno.test("spy function", () => {
     "function cannot be restored",
   );
   assertEquals(func.restored, false);
+
+  // Check if the returned type is correct:
+  const explicitTypesSpy = spy(point, "explicitTypes");
+  assertThrows(() => {
+    assertSpyCall(explicitTypesSpy, 0, {
+      // @ts-expect-error Test if passing incorrect argument types causes an error
+      args: ["not a number", "string"],
+      // @ts-expect-error Test if passing incorrect return type causes an error
+      returned: "not a boolean",
+    });
+  });
```

Comment:
Could you add a call that is valid with an assertSpyCall is valid to show it won't give typescript errors when you pass expected args and returned types? Currently there are only tests for when it catches typescript errors.

Maybe something like.

```ts
explicitTypesSpy(1, "hello");
assertSpyCall(explicitTypesSpy, 0, {
  args: [1, "hello"],
  returned: true,
});
```

Same for stub.